src/ServantElm.hs

```diff
@@ -166,6 +229,42 @@ mkRequest request =
 
     expect =
       case request ^. reqReturnType of
-        Just _ ->
-          "Http.expectWhatever toMsg"
+        Just elmTypeExpr ->
+          "Http.expectJson toMsg" <+> (typeRefDecoder . definitionToRef) elmTypeExpr
         Nothing -> error "mkHttpRequest: no reqReturnType?"
+
+    body =
+      case request ^. reqBody of
+        Just _ ->
+          "Http.jsonBody bodyValue"
+        Nothing ->
+          "Http.emptyBody"
+
+typeRefDecoder :: TypeRef -> Doc ann
+typeRefDecoder (RefCustom TypeName {..}) = "decode" <> pretty (T.takeWhile (/= ' ') unTypeName)
+typeRefDecoder (RefPrim elmPrim) = case elmPrim of
+  ElmUnit -> "Json.Decode.map (always ()) (Json.Decode.list Json.Decode.string)"
+  ElmNever -> "Json.Decode.fail \"Never is not possible\""
+  ElmBool -> "Json.Decode.bool"
+  ElmChar -> "elmStreetDecodeChar"
+  ElmInt -> "Json.Decode.int"
+  ElmFloat -> "Json.Decode.float"
+  ElmString -> "Json.Decode.string"
+  ElmTime -> "Iso.decoder"
+  ElmMaybe t ->
+    "maybe"
+      <+> typeRefDecoder t
+  ElmResult l r ->
+    parens "elmStreetDecodeEither"
+      <+> typeRefDecoder l
+      <+> typeRefDecoder r
```

Comment:
This seems wrong, you're only putting the parentheses around `elmStreetDecodeEither`, not around the entire expression.

Reply:
done